sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -536,3 +537,9 @@ def close(self):
         """
         self._close_handler()
         self._shutdown.set()
+        try:
+            self._service_bus_client._handlers.remove(self)  # pylint: disable=protected-access
```

Comment:
We really don't want any protected access here. It should be safe to assume that no-one outside of the current class touches an underscored name. We've been burn before when refactoring code. In this case, it is especially true because we are also catching `AttributeError`s that may happen if someone were to rename `_handlers` or similar.

Reply:
hey @johanste , any comments on the callback approach?

Reply:
per OOB discussion, use weakSet which could automatically discard elements for us